File: javaapp/src/main/math/raindrops/Raindrops.java
package main.math.raindrops;

/**
 * Package: PACKAGE_NAME
 * Created by lusinabrian on 31/07/16 or 31 Jul of 2016,
 * Time: 08:45.
 * Project: math.raindrops
 */
public class Raindrops {

    public static String convert(int number) {
        String rain = "";

        if (number % 3 == 0) {
            rain += "Pling";
        }

        if (number % 5 == 0) {
            rain += "Plang";
        }

        if (number % 7 == 0) {
            rain += "Plong";
        }

        if (rain.isEmpty()) {
            rain = Integer.toString(number);
        }

        return rain;
    }/*METHOD END*/

/*CLASS END*/
}

File: app/src/main/kotlin/com/kotlinapp/ktmath/raindrops/Raindrops.kt
package com.kotlinapp.ktmath.raindrops

/**
 * @author lusinabrian on 06/12/17.
 */
object Raindrops {
    fun convert(number: Int) : String{
        var output = ""
        (1..number)
                .asSequence()
                .filter { number % it == 0 }
                .forEach {
                    when(it){
                        3 -> output += "Pling"
                        5 -> output += "Plang"
                        7 -> output += "Plong"
                    }
                }
        if (output.isEmpty()){
            output = number.toString()
        }

        return output
    }

    fun convertUsingBuildStr(number: Int) = buildString {
        if (number.hasFactor(3)) append("Pling")
        if (number.hasFactor(5)) append("Plang")
        if (number.hasFactor(7)) append("Plong")
        if (isBlank()) append(number.toString())
    }

    private fun Int.hasFactor(factor : Int) = this.rem(factor) == 0
}